core/src/main/java/io/questdb/griffin/engine/groupby/DistinctRecordCursorFactory.java

```diff
@@ -72,13 +73,11 @@ public RecordCursor getCursor(SqlExecutionContext executionContext) {
         dataMap.clear();
         final RecordCursor baseCursor = base.getCursor(executionContext);
         try {
-            long maxInMemoryRows = executionContext.getCairoSecurityContext().getMaxInMemoryRows();
-            if (maxInMemoryRows > baseCursor.size()) {
-                dataMap.setMaxSize(maxInMemoryRows);
-                cursor.of(baseCursor, dataMap, mapSink);
-                return cursor;
-            }
-            throw LimitOverflowException.instance(maxInMemoryRows);
+            SqlResourceLimiter resourceLimiter = executionContext.getResourceLimiter();
+            resourceLimiter.checkLimits(baseCursor.size());
+            dataMap.setResourceLimiter(resourceLimiter);
```

Comment:
Execution context is bound to thread, so it resource limiter by extension. The RecordCursorFactory and its map is not thread bound. There is strong possibility that Thread A will be using resource limited of Thread B. I'm not sure if there is an issue right now doing this, but this potentially opens an unwanted can of worms in the future

Reply:
Surely this problem applies to the map as well